### ISSUE TITLE
Room integration

### ISSUE THREAD

**Issue by sleepylee** [nhphong/nimble-surveys#9]
I'm not sure if I understand your intention right, but at your [SurveyDAO](https://github.com/nhphong/nimble-surveys/blob/master/app/src/main/java/com/nhphong/nimblesurveys/data/db/SurveyDAO.kt) and its final usage on binding the data back to ViewModel, it looks cumbersome to me when we have to convert it first with room-rxjava and then at the end, without any transformation, the data is set back to a LiveData source. Could you please elaborate if you find this is useful somehow in your experience? I'd love to know more about this, because imo, Room has exposed interfaces to interact with LiveData at its nature.

**Comment by luongvo** [comment]
My suggestion is we could implement LiveData directly on SurveyDAO class. How is your thinking? 🤔

**Comment by nhphong** [comment]
If I understand correctly, having Room DAO returns a LiveData will make the flow look like this:
```Room Database -> DAO -> Repository -> ViewModel -> View```

So the View will subscribe directly to a LiveData object produced by the DAO. 
It means that if the field is changed in the database, it will notify the View immediately. (sounds like a real-time database)

However it is not what I wanted. What I wanted was having the ViewModel as a single source of truth that emits data on its purpose. For example, in this case, I want the ViewModel to emit data from the database only when the app is opened.

**Comment by nhphong** [comment]
@sleepylee @luongvo Feel free to correct me if my thinking was wrong.

**Comment by sleepylee** [comment]
@nhphong actually we're aligned on the idea of keeping the viewmodel as the single source of truth, what we meant is that you still can publish the changes to ViewModel's source not directly to the View, but instead of converting Room output source to Rx wrapper, we could convert it directly to LiveData type no? 🤔
 https://link.medium.com/vXCbThQrNV

**Comment by nhphong** [comment]
I've read the link you shared. It is similar to the flow I mentioned before:
```Room Database -> DAO -> Repository -> ViewModel -> View```

The View (`MainActivity`) subscribes to `ViewModel.allWords`
![image](https://user-images.githubusercontent.com/7122485/55949156-2cdea680-5c7c-11e9-97b5-0b2a3f0926ab.png)


The ViewModel gets that LiveData from Repository
![image](https://user-images.githubusercontent.com/7122485/55949187-408a0d00-5c7c-11e9-9970-cfe6f686e717.png)

The Repository, in turn, get the LiveData object from DAO
![image](https://user-images.githubusercontent.com/7122485/55949322-834be500-5c7c-11e9-9313-8aa02b2f148e.png)

In the end, the View will subscribe directly to a LiveData object produced by the DAO

**Comment by sleepylee** [comment]
well..if you want to have a separate observable source on ViewModel, you can subscribe to the DAO source as well no? Just to confirm that your very first approach with rxjava is correct as well, but instead of subscribe via a Rx source, my point is that we can observe directly from the LiveData source no? 🤔

**Comment by nhphong** [comment]
Right. We can create an LiveData instance within the ViewModel, and let it subscribe to the one created by the DAO.

However, if we follow that way, it will end up with a Repository like this
![image](https://user-images.githubusercontent.com/7122485/55953929-9912d780-5c87-11e9-92df-09ab47bf8991.png)

A mix between RxObservables and LiveData in the interface of a Repository. 🤔 🤔 
I think it would be harder to read. How do you think?

**Comment by sleepylee** [comment]
I agree it's not easy when mixing both techniques in one, but indeed that's our concerns since the beginning 😄so, the question now is: can you propose a viable solution for only LiveData implementations to the rest 2 methods there for consistency?

**Comment by nhphong** [comment]
Since LiveData is just an observable source, converting it to/from RxObservable is not a difficult task. What I'm concern here is whether we should do that. 
We already have a nice `RxJava2CallAdapterFactory` for Retrofit, but I never see people trying to implement a `LiveData` adapter for it (or at least, not yet 😅).

If the implementation is just for this technical challenge, we can give it a try though.

**Comment by nhphong** [comment]
On the second thought, converting RxJava to LiveModel in this case takes time and effort, and it's not natural for the case of Retrofit. 
Since we already have a nice support for RxJava from Room, I think it's better that we keep this way for simplicity, well, unless you insist.
